### PR TITLE
Make the hangup button do things for conference calls

### PR DESCRIPTION
**Pull request by turt2live**
**Design reviewer**: this doesn't have an easy adhoc build, sorry - DM me for details of one though. This PR is looking for your grace to enter mainline Element.

----

**Review with https://github.com/vector-im/element-web/pull/15219**

Behaviour constraints:
* If you're not in the conference, use a grey button that does nothing.
* If you're in the conference, show a button:
  * If you're able to modify widgets in the room, annotate it in the context of ending the call for everyone and remove the widget. Use a confirmation dialog.
  * If you're not able to modify widgets in the room, hang up.

For this we know that persistent Jitsi widgets will mean that the user is in the call, so we use that to determine if they are actually participating.

Screenshots:
image
image
image

### CODE REVIEW

File: src/CallHandler.js [hunk 1]
```diff
@@ -444,6 +429,40 @@ async function _startCallApp(roomId, type) {
...
+            const jitsiWidgets = roomInfo.widgets.filter(w => WidgetType.JITSI.matches(w.type));
+            jitsiWidgets.forEach(w => {
+                // setting invalid content removes it
+                WidgetUtils.setRoomWidget(roomId, w.id);
+            });
+        },
+    });
+}
+
+function _hangupWithCallApp(roomId) {
```
Comment (jryans) [inline src/CallHandler.js:453]: Maybe just "hangup"...?

```suggestion
function _hangupCallApp(roomId) {
```

File: src/WidgetMessaging.js [hunk 2]
```diff
@@ -107,6 +107,17 @@ export default class WidgetMessaging {
         });
     }
 
+    /**
+     * Tells the widget to hang up on its call.
+     * @returns {Promise<*>} Resolves when teh widget has acknowledged the message.
```
Comment (jryans) [inline src/WidgetMessaging.js:112]:
```suggestion
     * @returns {Promise<*>} Resolves when the widget has acknowledged the message.
```

**Review by jryans — APPROVED** [review]

File: src/CallHandler.js (hunk 1 above)
Comment (turt2live) [reply src/CallHandler.js]: applied manually due to renaming

**Comment by niquewoodhouse** [comment]
`Ending the conference will end the call for everyone...` could be `This will end the conference for everyone...` to simplify the copy.

**Review by niquewoodhouse — APPROVED** [review]
LGTM `Ending the conference will end the call for everyone...` could be `This will end the conference for everyone...` to simplify the copy.

**Comment by turt2live** [comment]
tests are failing due to an old branch - will monitor develop after merging

**Pull request merged**